### DISCUSSION TITLE
Calculation of via point data

### DISCUSSION THREAD

**Discussion by MotoUKRider** [devemux86/cruiser#692] [Q&A] [answered]
Is it possible to apply the "intermediate" distance/timing information to a newly toggled ViaPoint in the Waypoints list?
At present, you only see the "total so far" line when toggling a ShapingPoint as a ViaPoint, so you have to toggle it, confirm it and re-enter the list view in order to see the 2nd line of info.

Here's an example from Desktop:
![cruiser-waypoint-timings](https://github.com/user-attachments/assets/50a9080f-21f9-4939-a6df-1edbd4912734)

This would make it easier to see the gap between each Via "on the fly" without tapping the route on the map to look at each point.

**Comment by devemux86** [comment, marked as answer]
@MotoUKRider Thanks for the suggestion.

However, such data can only be calculated after finalizing the waypoints and the route, like in 2nd image.

**Reply by MotoUKRider** [reply]
@devemux86 Hmmm, OK.  But the data is already visible in the first image - all it has to do, upon the toggle action, is to seek to the previous and next ViaPoints to ascertain the distance, eg: 55mi, 60mi, 65.6mi to give 5mi and 5.6mi plus the durations.

**Reply by devemux86** [reply]
This data already exists in the route, it is not calculated in real time.
Changing this would require significant changes to many elements.

**Reply by devemux86** [reply]
Also it is more complicated.

If we reorder some waypoints, then the distances / times are all wrong.
So it is better to remove them from this screen to avoid such confusion.

**Reply by MotoUKRider** [reply]
Hmmm. I've always liked the summary of distance+time in that list, especially as you can't touch the route and see the status at any part of the line.

I would understand the removal of the information text *only* after an edit operation (re-arrange, delete) on that dialog, but to completely lose it would be unhelpful.

**Reply by devemux86** [reply]
> the removal of the information text only after an edit operation (re-arrange, delete)

Reorder, delete, change type, it quickly becomes complicated to manage.
Also then users will ask why this info suddenly disappears.

> you can't touch the route and see the status at any part of the line.

You can tap the via points on the map and see their detailed distance / time.

**Reply by MotoUKRider** [reply]
What if the route is A to B and I want to find a place along it, after 90 mins, to rest?

The current press+hold on the route acts as a "whole" route action so I really think there should be some way to easily see where/when you are on a route via an action regardless of the info text being scrapped. This would be especially useful on unknown roads where 50mi on a mountain pass would be a longer (and more challenging) ride than a 50mi motorway sprint.

Otherwise, it's place a Via, no, too soon. Move the map, place another, oops too far.  *repeat to fade*

**Reply by devemux86** [reply]
When long press on route line we could calculate approximately the distance / time at the pressed point
and we could probably display them at the route menu title instead of current route total distance / time.

@MotoUKRider However this is a different feature request. If you mean this, please create a new thread,
so we will not forget it here.
